plugins/sqlfluff-templater-dbt/test/templater_test.py

```diff
@@ -296,6 +311,49 @@ def test__templater_dbt_handle_exceptions(
     assert violations[0].desc().replace("\\", "/").startswith(exception_msg)
 
 
+def test__templater_dbt_handle_database_connection_failure(
+    project_dir, dbt_templater  # noqa: F811
+):
+    """Test the result of a failed database connection."""
+    from dbt.adapters.factory import get_adapter
+
+    src_fpath = "plugins/sqlfluff-templater-dbt/test/fixtures/dbt/error_models/exception_connect_database.sql"
+    target_fpath = os.path.abspath(
+        os.path.join(
+            project_dir, "models/my_new_project/exception_connect_database.sql"
+        )
+    )
+    # We move the file that throws an error in and out of the project directory
+    # as dbt throws an error if a node fails to parse while computing the DAG
+    os.rename(src_fpath, target_fpath)
+    try:
+        _, violations = dbt_templater.process(
+            in_str="",
+            fname=target_fpath,
+            config=FluffConfig(configs=DBT_FLUFF_CONFIG),
+        )
+    except Exception as e:
+        if DBT_VERSION_TUPLE == (1, 0):
```

Comment:
Seems like this might be more "future proof":
```suggestion
        if DBT_VERSION_TUPLE >= (1, 0):
```